### PR TITLE
DOC: Update array.rst - Closes Issue #22949

### PR DESCRIPTION
**Pull request by arunkumarkota**
The "2" link has no existing target in the second paragraph at https://numpy.org/doc/1.24/reference/c-api/array.html#array-iterators

This is wrt to issue: https://github.com/numpy/numpy/issues/22949

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
It's probably better to update the link rather than remove it entirely. From context I think this was intended to point to the [array iterator](https://numpy.org/doc/1.24/reference/c-api/iterator.html#array-iterator) section in the Array Iterator API doc.

**Comment by arunkumarkota** [comment]
> It's probably better to update the link rather than remove it entirely. From context I think this was intended to point to the [array iterator](https://numpy.org/doc/1.24/reference/c-api/iterator.html#array-iterator) section in the Array Iterator API doc.

update the PR. pls review

File: doc/source/reference/c-api/array.rst [hunk 1]
```diff
@@ -2476,7 +2476,7 @@ the new array iterator, :c:type:`NpyIter`.
 
 An array iterator is a simple way to access the elements of an
 N-dimensional array quickly and efficiently. Section `2
-<#sec-array-iterator>`__ provides more description and examples of
+<iterator.html>`__ provides more description and examples of
 this useful approach to looping over an array.
```
Comment (mattip) [inline doc/source/reference/c-api/array.rst:2480]: Maybe replace this with
```
N-dimensional array quickly and efficiently, as seen in `the
example <iteration-example>` which provides more description
of this useful approach to looping over an array from C.
```

Comment (arunkumarkota) [reply doc/source/reference/c-api/array.rst]: updated PR. Pls review

**Comment by mattip** [comment]
Close, but the [new link](https://output.circle-artifacts.com/output/job/0e3b0ffe-1089-41fe-90e4-7acbf6a840eb/artifacts/0/doc/build/html/reference/c-api/array.html#array-iterators) is broken. I am not sure of the exact sphinx incantation to get [this link](https://output.circle-artifacts.com/output/job/0e3b0ffe-1089-41fe-90e4-7acbf6a840eb/artifacts/0/doc/build/html/reference/c-api/iterator.html#iteration-example)

**Comment by rossbar** [comment]
> I am not sure of the exact sphinx incantation to get [this link](https://output.circle-artifacts.com/output/job/0e3b0ffe-1089-41fe-90e4-7acbf6a840eb/artifacts/0/doc/build/html/reference/c-api/iterator.html#iteration-example)

This is always the tricky part. Looking at numpy's intersphinx inventory it seems the section in question isn't included as a link target, so one would have to be explicitly created in https://github.com/numpy/numpy/blob/main/doc/source/reference/c-api/iterator.rst. This could be done by adding e.g. `.. _iteration-example:` above the section header, then using `<iteration-example>` as the target in the link role. I'd push up this suggestion myself, but I am unable to do so since the PR is originating from your main branch @arunkumarkota

**Comment by rossbar** [comment]
This is very nearly there - just needs a fix to the link. I've added the necessary change locally, but am unable to push to your main branch. I will close this and re-open in a follow-up PR just to get this last little change in.

**Pull request closed without merging**